packages/core/src/data-grid/data-grid-lib.ts

```diff
@@ -510,11 +538,20 @@ function roundedRect(
     ctx.arcTo(x, y, x + radius.tl, y, radius.tl);
 }
 
-export function drawBoolean(args: BaseDrawArgs, data: boolean, canEdit: boolean) {
+export function drawBoolean(args: BaseDrawArgs, data: boolean | null | undefined, canEdit: boolean) {
+    if (!canEdit && data === null) {
+        return;
+    }
+
     const { ctx, hoverAmount, theme, x, y, w, h, highlighted, hoverX, hoverY } = args;
+
     const hoverEffect = 0.35;
 
-    ctx.globalAlpha = canEdit ? 1 - hoverEffect + hoverEffect * hoverAmount : 0.4;
+    let alpha = canEdit ? 1 - hoverEffect + hoverEffect * hoverAmount : 0.4;
+    if (data === null) {
+        alpha *= hoverAmount;
+    }
+    ctx.globalAlpha = alpha;
```

Comment:
if alpha === 0 return early. Let's not waste the draw calls drawing nothing. Unfortunately the canvas API will faithfully perform these useless operations.

Reply:
Ooohhh good call. Missed that one.